site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/HapiFhirRepository.java

```diff
@@ -38,49 +38,89 @@
      */
     public String saveOrganization(Organization organization) {
         // Log the request
-        logger.info(FhirRepo.SAVE_REQUEST.message(),
-                fhirContextWrapper.prettyPrint(organization));
+        if (logger.isInfoEnabled()) {
+            logger.info(FhirRepo.SAVE_REQUEST.message(),
+                    fhirContextWrapper.prettyPrint(organization));
+        }
 
         Bundle responseBundle;
         try {
             responseBundle = fhirContextWrapper.executeTrasaction(fhirUri, 
                 bundle(organization, ORGANIZATION_ENTITY_NAME));
         } catch (BaseServerResponseException e) {
-            logger.warn(FhirRepo.UPDATE_ERROR.message(), e);
+            if (logger.isWarnEnabled()) {
+                logger.warn(FhirRepo.UPDATE_ERROR.message(), e);
+            }
             throw new RestException(e.getMessage(), e);
         }
         IBaseResource responseElement = extractResponseResource(responseBundle);
 
         // Log the response
-        logger.info(FhirRepo.SAVE_RESPONSE.message(),
+        if (logger.isInfoEnabled()) {
+            logger.info(FhirRepo.SAVE_RESPONSE.message(),
                     fhirContextWrapper.prettyPrint(responseElement));
+        }
 
         return responseElement.getIdElement().getIdPart();
     }
 
+    /**
+     * @param name of the organization to search.
+     * @return id of the saved organization
+     */
+    public String findOrganizationByName(String name) {
+        // Log the request
+        if (logger.isInfoEnabled()) {
+            logger.info(FhirRepo.SAVE_REQUEST.message(), name);
+        }
+        Bundle responseBundle = new Bundle();
+
+        try {
+            responseBundle = fhirContextWrapper.executeSearchByName(fhirUri, name);
+
+        } catch (BaseServerResponseException e) {
+            if (logger.isWarnEnabled()) {
+                logger.warn(FhirRepo.SEARCH_ERROR.message(), e);
+            }
+            throw new RestException(e.getMessage(), e);
+        }
+
+        if (!responseBundle.getEntry().isEmpty()) {
+            IBaseResource responseElement = extractResponseResource(responseBundle);
+            return responseElement.getIdElement().getIdPart();
+        }
+
+        return FhirRepo.NOT_FOUND.message();
```

Comment:
I think we should think what is the appropriate action here... Perhaps a not found exception or maybe an empty org, or null.
@darrengsmithson what do you think?

Reply:
We are returning null if not found.
As per the requirement, we return an exception, if Site Name Already Exists